src/datachain/cli/commands/storage/local.py

```diff
@@ -0,0 +1,127 @@
+from typing import TYPE_CHECKING
+
+from datachain.cli.commands.storage.base import StorageImplementation
+from datachain.cli.commands.storage.utils import build_file_paths, validate_upload_args
+from datachain.error import DataChainError
+
+if TYPE_CHECKING:
+    from datachain.client.fsspec import Client
+
+
+class LocalStorageImplementation(StorageImplementation):
+    def upload_to_remote(self, source_cls: "Client", destination_cls: "Client"):
+        from tqdm import tqdm
+
+        source_fs = source_cls.create_fs()
+        destination_fs = destination_cls.create_fs()
+        is_dir = validate_upload_args(
+            self.args.source_path, self.args.recursive, source_fs
+        )
+        file_paths = build_file_paths(
+            self.args.source_path, self.args.destination_path, source_fs, is_dir
+        )
+        destination_bucket, _ = destination_cls.parse_url(self.args.destination_path)
+
+        for dest_path, source_path in file_paths.items():
+            file_size = source_fs.info(source_path)["size"]
+            print(f"Uploading {source_path} to {dest_path}")
+
+            with tqdm(total=file_size, unit="B", unit_scale=True) as pbar:
+                with source_fs.open(source_path, "rb") as source_file:
+                    with destination_fs.open(
+                        f"{destination_bucket}/{dest_path}", "wb"
+                    ) as dest_file:
+                        while True:
+                            chunk = source_file.read(8192)
+                            if not chunk:
+                                break
+                            dest_file.write(chunk)
+                            pbar.update(len(chunk))
+
+        self.save_upload_logs(self.args.destination_path, file_paths, source_fs)
+
+    def download_from_remote(self, destination_cls: "Client"):
```

Comment:
remote == cloud in this case?

let's use `cloud` then ... `remote` is used for too many things now, and we should stop using it for all of them really

Reply:
Updated.